### PR TITLE
feat: support loading debug urls with loadURL()

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This allows loading the following [browser-side debug URLs](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/debug_urls.cc;l=131;drc=be4c5173c2566193f22111961e7f528e4edaa18a) via `webContents.loadURL()`:

- chrome://inducebrowsercrashforrealz/
- chrome://uithreadhang/
- chrome://delayeduithreadhang/
- chrome://gpuclean/
- chrome://gpucrash/
- chrome://gpuhang/
- chrome://memory-pressure-critical/
- chrome://memory-pressure-moderate/

on Windows:

- chrome://inducebrowserheapcorruption/

when building with ASan:

- chrome://browser-heap-overflow/
- chrome://browser-heap-underflow/
- chrome://browser-use-after-free/

when building with ASan on Windows:

- chrome://browser-corrupt-heap-block/
- chrome://browser-corrupt-heap/

There also exist several [renderer-side debug URLs](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/common/chrome_debug_urls.cc;l=166;drc=be4c5173c2566193f22111961e7f528e4edaa18a),
which are handled on the renderer side and are not affected by this change.
(They already worked.)

These URLs will only work when loaded explicitly using `loadURL`. Web content which navigates to one of these URLs through `location.href` or otherwise will not trigger the handler.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added support for debug URLs such as `chrome://gpucrash`.

### CODE REVIEW

File: shell/browser/api/electron_api_web_contents.cc [hunk 1]
```diff
@@ -1978,7 +1978,8 @@ void WebContents::LoadURL(const GURL& url,
   // Calling LoadURLWithParams() can trigger JS which destroys |this|.
   auto weak_this = GetWeakPtr();
 
-  params.transition_type = ui::PAGE_TRANSITION_TYPED;
+  params.transition_type = ui::PageTransitionFromInt(
+      ui::PAGE_TRANSITION_TYPED | ui::PAGE_TRANSITION_FROM_API);
```
Comment (deepak1556) [inline shell/browser/api/electron_api_web_contents.cc:1982]: What is the reasoning for adding the transition type `ui::PAGE_TRANSITION_FROM_API` ?

Comment (nornagon) [reply shell/browser/api/electron_api_web_contents.cc:1982]: It probably is sufficient, but FROM_API seemed more semantically correct. The docs say FROM_API is reserved for embedders but it does seem to have some meaning to content:

https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigation_controller_impl.cc;l=179;drc=4f5373794ada7eba6da97bff0229fb0d85f601ec

the default impl of `IsExplicitNavigation` checks for `ui::PAGE_TRANSITION_FROM_ADDRESS_BAR`. I'm not sure whether it's correct to include that here.

Comment (deepak1556) [reply shell/browser/api/electron_api_web_contents.cc:1982]: Browser navigations code paths from `headless`, `shell` components seem to use `ui::PAGE_TRANSITION_FROM_ADDRESS_BAR` along with the typed core transition type, so it should be safe to use here as well.

Comment (nornagon) [reply shell/browser/api/electron_api_web_contents.cc:1982]: Alrighty, I'll switch this to use `ui::PAGE_TRANSITION_TYPED | ui::PAGE_TRANSITION_FROM_ADDRESS_BAR`.

**Review by deepak1556 — APPROVED** [review]

**Pull request merged**